csrc/quantization/cutlass_w8a8/scaled_mm_c2x.cu

```diff
@@ -134,6 +134,46 @@ struct ScaledEpilogue {
   }
 };
 
+template <typename ElementD, typename OutputTileThreadMap>
+struct ScaledEpilogueAzp
+    : private ScaledEpilogue<ElementD, OutputTileThreadMap> {
```

Comment:
Does it make sense for `ScaledEpilogueAzp` to be a subclass of `ScaledEpilogue`?

I think if the types were more compositional than they are and if `ScaledEpilogue` remained encapulsated, it would. In this case though, `ScaledEpilogueAzp` really cracks open `ScaledEpilogue` and looks inside of it -- the fact that you're using `cutlass::multiply_add` to apply both the bias and one of the scales necessitates this I think.

In this case, I would suggest not having `ScaledEpilogueAzp` inherit from `ScaledEpilogue` and instead factoring out the ScaleA and ScaleB type definitions another way. Maybe we could add a more generic type called `EpilogueWithTensorExpansion` and define the types there?

Reply:
I like that! I'll try it, and if too complex, I'll just repeat them